Flatcar VM bootstrap fails

The CI test fails across all branches: https://github.com/kubernetes-sigs/security-profiles-operator/actions/runs/3680636245/jobs/6232228689

```
    default: [kubelet-check] It seems like the kubelet isn't running or healthy.
    default: [kubelet-check] The HTTP call equal to 'curl -sSL http://localhost:10248/healthz' failed with error: Get "http://localhost:10248/healthz": dial tcp [::1]:10248: connect: connection refused.
    default: 
    default: Unfortunately, an error has occurred:
    default: 	timed out waiting for the condition
    default: 
    default: This error is likely caused by:
    default: 	- The kubelet is not running
    default: 	- The kubelet is unhealthy due to a misconfiguration of the node in some way (required cgroups disabled)
    default: 
    default: If you are on a systemd-powered system, you can try to troubleshoot the error with the following commands:
    default: 	- 'systemctl status kubelet'
    default: 	- 'journalctl -xeu kubelet'
    default: 
    default: Additionally, a control plane component may have crashed or exited when started by the container runtime.
    default: To troubleshoot, list all containers using your preferred container runtimes CLI.
    default: Here is one example how you may list all running Kubernetes containers by using crictl:
    default: 	- 'crictl --runtime-endpoint unix:///run/containerd/containerd.sock ps -a | grep kube | grep -v pause'
    default: 	Once you have found the failing container, you can inspect its logs with:
    default: 	- 'crictl --runtime-endpoint unix:///run/containerd/containerd.sock logs CONTAINERID'
    default: error execution phase wait-control-plane: couldn't initialize a Kubernetes cluster
    default: To see the stack trace of this error execute with --v=5 or higher
The SSH command responded with a non-zero exit status. Vagrant
assumes that this means the command failed. The output for this command
should be in the log above. Please read the output to determine what
went wrong.
make: *** [vagrant-up-flatcar] Error 1
```

--- a/hack/ci/Vagrantfile-flatcar
+++ b/hack/ci/Vagrantfile-flatcar
@@ -2,9 +2,10 @@
 # vi: set ft=ruby :
 
 Vagrant.configure("2") do |config|
-  config.vm.box = "flatcar-lts"
+  config.vm.box = "flatcar-main"
   config.vm.box_check_update = true
-  config.vm.box_url = "https://lts.release.flatcar-linux.net/amd64-usr/current/flatcar_production_vagrant.box"
+  #config.vm.box_url = "https://stable.release.flatcar-linux.net/amd64-usr/current/flatcar_production_vagrant.box"
+  config.vm.box_url = "http://bincache.flatcar-linux.net/images/amd64/9999.0.0+tormath1-ignition-vagrant/flatcar_production_vagrant.box"
   memory = 8192
   cpus = 4
 
@@ -24,12 +25,14 @@ Vagrant.configure("2") do |config|
     v.cpus = cpus
   end
 
+  config.ssh.connect_timeout = 30
   config.ssh.username = 'core'
   config.ssh.forward_agent = true
   config.ssh.insert_key = true
   config.ssh.keep_alive = true
 
   config.vm.synced_folder ".", "/vagrant", type: "rsync"
+  config.vm.boot_timeout = 600
 
   config.vm.provision "install-dependencies", type: "shell", run: "once" do |sh|
     sh.inline = <<~SHELL
@@ -90,7 +93,7 @@ Vagrant.configure("2") do |config|
       # Install tools
       curl -sSL -o $DOWNLOAD_DIR/jq "https://github.com/stedolan/jq/releases/download/jq-1.6/jq-linux64"
       chmod +x $DOWNLOAD_DIR/jq
-      curl -sSL -o $DOWNLOAD_DIR/flatcar_developer_container.bin.bz2 "https://lts.release.flatcar-linux.net/amd64-usr/current/flatcar_developer_container.bin.bz2"
+      curl -sSL -o $DOWNLOAD_DIR/flatcar_developer_container.bin.bz2 "https://stable.release.flatcar-linux.net/amd64-usr/current/flatcar_developer_container.bin.bz2"
       bzcat $DOWNLOAD_DIR/flatcar_developer_container.bin.bz2 > $DOWNLOAD_DIR/flatcar_developer_container.bin
 
       curl -sSL "https://github.com/containernetworking/plugins/releases/download/${CNI_VERSION}/cni-plugins-linux-amd64-${CNI_VERSION}.tgz" | tar -C /opt/cni/bin -xz
@@ -125,7 +128,7 @@ Vagrant.configure("2") do |config|
       kubectl taint nodes --all node-role.kubernetes.io/control-plane-
 
       # Install CNI
-      kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.24.0/manifests/tigera-operator.yaml
+      kubectl create -f https://raw.githubusercontent.com/projectcalico/calico/v3.24.5/manifests/tigera-operator.yaml
       kubectl apply -f /vagrant/hack/ci/flatcar-cni-plugin.yaml
     SHELL
   end

--- a/hack/ci/e2e-flatcar.sh
+++ b/hack/ci/e2e-flatcar.sh
@@ -38,6 +38,9 @@ export PATH="$HOSTFS_DEV_MOUNT_PATH/opt/bin:$PATH"
 export KUBECONFIG=$HOSTFS_DEV_MOUNT_PATH/etc/kubernetes/admin.conf
 alias k=kubectl
 
+# Configure git to consider the mounted host filesystem as safe
+git config --global --add safe.directory "/hostfs/vagrant"
+
 if "${E2E_TEST_FLAKY_TESTS_ONLY}"; then
     make test-flaky-e2e
 else

